### DISCUSSION TITLE
How to create a package from generated code by typegraphql-prisma?

### DISCUSSION THREAD

**Discussion by jsefiani** [MichalLytek/typegraphql-prisma#65] [Q&A]
I'm using Yarn 2 Workspaces to manage multiple sub-projects/packages in a mono-repo with the following folder structure.

📂 apps
 -┣ 📦 api
 -┣ 📦 portal
 -┣ 📦 engine
📂 shared
 -┣ 📦 database
   --┗ 📂 prisma
   --┗ 📂 src 
      ---┗  📂 type-graphql // custom-configured output folder (contains generated code by typegraphql-prisma)
      ---┗ index.ts // export * from './type-graphql'
   --┗ tsonfig.json
   --┗ babel.config.js
   --┗ package.json
 -┣ 📦 ui
 -┣ 📦 utils
...
The database package should export all generated code by typegraphql-prisma so that I can use it in the api package (e.g. import { Advertiser, Brand } from '@monorepo/database').

When I do it this way and use the package database in the api package, I get the following error:
Cannot find module './type-graphql'

So I'm wondering what I'm doing wrong.

This is my babel file:
```
const BABEL_ENV = process.env.BABEL_ENV;
const isCommonJS = BABEL_ENV !== undefined && BABEL_ENV === 'cjs';
const isESM = BABEL_ENV !== undefined && BABEL_ENV === 'esm';

module.exports = function (api) {
	api.cache(true);

	const presets = [
		[
			'@babel/env',
			{
				loose: true,
				modules: isCommonJS ? 'commonjs' : false,
				targets: {
					esmodules: isESM ? true : undefined,
				},
			},
		],
		'@babel/preset-typescript',
	];

	const plugins = [
		'@babel/plugin-transform-runtime',
		'babel-plugin-transform-typescript-metadata',
		['@babel/plugin-proposal-decorators', { legacy: true }],
		'babel-plugin-parameter-decorator',
		['@babel/plugin-proposal-class-properties', { loose: true }],
		'@babel/plugin-transform-classes',
		'@babel/plugin-transform-typescript',
	];

	return {
		presets,
		plugins,
	};
};
```

Prisma snippet:
```
generator typegraphql {
  provider           = "typegraphql-prisma"
  output             = "../src/type-graphql"
  emitTranspiledCode = true
}
```
Any thoughts? Tips?

Please let me know if something is unclear.

**Comment by MichalLytek** [comment]
`typegraphql-prisma` generates source code, so it's the same code you would write by yourself.

So I think there's nothing special required for monorepo 🤔

**Reply by MichalLytek** [reply]
> emitTranspiledCode = true

That's your issue with the setup as babel is transpiling this files I believe. Start with normal TS source code.

**Reply by jsefiani** [reply]
Well, tried that but when I use the `database` package, I get the following error: 
`UnhandledPromiseRejectionWarning: Error: Unable to find type metadata for input type or object type named ''`

**Reply by jsefiani** [reply]
This is my tsconfig (database package):
```
{
	"extends": "../../tsconfig.json",
	"include": ["src"],
	"compilerOptions": {
		"target": "es2019",
		"lib": ["es2019", "esnext.asynciterable"],
		"outDir": "./dist",
		"rootDir": "./src",
		"jsx": "preserve",
		"module": "commonjs",
		"experimentalDecorators": true,
		"emitDecoratorMetadata": true,
		"strictPropertyInitialization": false,
		"noImplicitAny": false
	}
}
```

**Reply by MichalLytek** [reply]
You need to hoist `type-graphql` and `graphql` dependencies in your monorepo.

**Reply by jsefiani** [reply]
So I have to install those 2 packages in the root right? Instead of package level?

**Reply by MichalLytek** [reply]
https://github.com/lerna/lerna/blob/main/doc/hoist.md

**Reply by jsefiani** [reply]
Sorry, that didn't work either
